ignite/distributed/comp_models/base.py

```diff
@@ -271,8 +276,10 @@ def spawn(*args: Any, **kwargs: Any) -> None:
     def all_reduce(self, tensor: Union[torch.Tensor, Number], op: str = "sum") -> Union[torch.Tensor, Number]:
         return tensor
 
-    def all_gather(self, tensor: Union[torch.Tensor, Number]) -> Union[torch.Tensor, Number]:  # type: ignore
-        return tensor
+    def all_gather(
+        self, tensor: Union[torch.Tensor, Number, str]
+    ) -> Union[torch.Tensor, Number, List[Number], List[str]]:
+        return cast(Union[torch.Tensor, Number, List[Number], List[str]], tensor)
```

Comment:
Let's implement it as
```python
def all_gather(self, tensor: Union[torch.Tensor, Number, str]) -> Union[torch.Tensor, List[Number], List[str]]:
    if isinstance(tensor, torch.Tensor):
        return tensor
    return cast(Union[List[Number], List[str]], [tensor, ])
```

And a test may be required to make codecov and myself happy :)